### PR TITLE
xPaths ex

### PR DESCRIPTION
**Pull request by AlinaLazea**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "test:wdio": "wdio wdio.conf.js --spec bayutSearchBedsFilter.test.js"
+    "test:wdio": "wdio wdio.conf.js"
```
Comment (AdeelZahir) [inline package.json:8]: 👍

File: tests/pageobject/bayutPropertyPageSearchFilters.page.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
+import Page from "./page.js"
+
+class xPathFilters extends Page{
+    //element, type object, methods getText, etc
```
Comment (AdeelZahir) [inline tests/pageobject/bayutPropertyPageSearchFilters.page.js]: You can remove all the commented lines so it looks more clean

File: tests/pageobject/bayutPropertyPageSearchFilters.page.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        return $(`//*[contains(@title, "Bayut.com")]`)
+    }
+
+
+    // selector type string
+    /**
+     * out: //a[contains(@title, "Bayut.com")
+     */
+    get selectorBayutLogo(){
+        return $(`//a[contains(@title, "Bayut.com")]`).selector
```
Comment (AdeelZahir) [inline tests/pageobject/bayutPropertyPageSearchFilters.page.js]: Why do you place `.selector` in the end?

File: tests/pageobject/bayutPropertyPageSearchFilters.page.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+     */
+    get elementFunctionBayutLogo(){
+        return $(`//a[contains(@title, "Bayut.com")]`).element
+    }
+
+//-------------------------------------------------------------------------------------------------
+
+    // get the buy btn
+    get ebuyBtn(){
+        return $(`${this.eBayutLogo.selector}/following-sibling::div/span[1]`)
```
Comment (AdeelZahir) [inline tests/pageobject/bayutPropertyPageSearchFilters.page.js]: Could also be done like this.

```suggestion
        return $('//*[@aria-label="Mini search form"]//*[text()="For Sale"]')
```

This way, we have less dependability on other elements. We don't have to rely on `eBayutLogo` or the hierarchy in the DOM

File: tests/pageobject/bayutPropertyPageSearchFilters.page.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    }
+
+    get eRentBtn(){
+        return $(`${this.eBayutLogo.selector}/following-sibling::div/span[2]`)
+    }
+
+    get eblogBtn(){
+        return $('//*[@aria-label = "Blogs" and @title = "MyBayut"]')
```
Comment (AdeelZahir) [inline tests/pageobject/bayutPropertyPageSearchFilters.page.js:25]: 👍

File: tests/specs/bayutSearchBedsFilter.test.js [hunk 6]
```diff
@@ -1,9 +1,9 @@
-import Filters from "./pageobject/bayutSearchFilters.page.js"
-import Results from "./pageobject/bayutSearchResults.page.js"
+import Filters from "../pageobject/bayutSearchFilters.page.js"
+import Results from "../pageobject/bayutSearchResults.page.js"
 require('mocha-steps')
```
Comment (AdeelZahir) [inline tests/specs/bayutSearchBedsFilter.test.js]: We can skip this way of including `mocha-steps`.
Another way to do it is by mentioning it in wdio.conf under `mochaOpts`
image

File: tests/pageobject/bayutPropertyPageSearchFilters.page.js [hunk 7]
```diff
@@ -0,0 +1,201 @@
+import Page from "./page.js"
+
+class xPathFilters extends Page{
```
Comment (AdeelZahir) [inline tests/pageobject/bayutPropertyPageSearchFilters.page.js]: Can we have a meaningful name here? Maybe `SearchPageFilters`

File: tests/pageobject/bayutPropertyPageSearchFilters.page.js (hunk 4 above)
Comment (AlinaLazea) [reply tests/pageobject/bayutPropertyPageSearchFilters.page.js]: The element that you suggested doesn't exist. Instead i have change it with $('//*[text() = "Buy"]')

**Pull request merged**